fix(swift): Address all remaining "not implemented" errors

nmote opened the pull request:
There were just a handful of Swift constructs that still could lead to exceptions during the translation from the tree sitter CST to the generic AST.

This PR addresses these cases, mostly with the new `Raw_tree` functionality which will at least allow us to perform matching on these constructs.

Test plan: Automated tests

kopecs commented on languages/swift/generic/Parse_swift_tree_sitter.ml:
```diff
@@ -1750,9 +1765,10 @@ and map_key_path_postfixes (env : env) (x : CST.key_path_postfixes) =
+      let _args = Option.map (map_arguments env) v2 |> Common.optlist_to_list in
+      let _rb = (* "]" *) token env v3 in
+      (* TODO *)
+      lb
```
Is it better to do something like https://github.com/returntocorp/semgrep/pull/8845/files#diff-bc37eef52156d5f5900d5b5e1c601b0bd0c251a3725f7b60b2242093e1e0565dR2666-R2671 instead?

nmote replied:
The trouble is that then I would have to modify the other cases, since they all return a single `Tok.t`, as well as the callsites.

kopecs replied:
I see. This seems fine for now then.

kopecs commented on languages/swift/generic/Parse_swift_tree_sitter.ml:
```diff
@@ -1736,12 +1745,18 @@ and map_arguments (env : env) ((v1, v2) : CST.interpolation_contents) :
   v1 @ v2
 
 and map_key_path_component (env : env) (x : CST.key_path_component) =
+  (* NOTE: According to a note at the callsite, this doesn't end up getting
+   * called in practice. At some point. *)
```
Better to also add a comment there in case that ends up being modified? Probably okay as-is given return type being unit.

nmote replied:
There's already a comment there

kopecs replied:
Was thinking more so about this function not really being implemented, beyond just that it isn't currently called since the list is always empty.

nmote replied:
I see, I can do that

nmote committed:
fix(swift): Address all remaining "not implemented" errors

There were just a handful of Swift constructs that still could lead to
exceptions during the translation from the tree sitter CST to the
generic AST.

This PR addresses these cases, mostly with the new `Raw_tree`
functionality which will at least allow us to perform matching on these
constructs.

Test plan: Automated tests
